lib/buffer.js

```diff
@@ -317,8 +317,16 @@ Object.defineProperty(Buffer.prototype, 'offset', {
 function slowToString(encoding, start, end) {
   var loweredCase = false;
 
-  start = start >>> 0;
-  end = end === undefined || end === Infinity ? this.length : end >>> 0;
+  start = +start;
+  if (!Number.isInteger(start)) {
+    start = start >>> 0;
+  }
+
+  if (end === undefined || end === Infinity) {
+    end = this.length;
+  }
+  end = +end;
+  end = end < 0 ? this.length : end >>> 0;
```

Comment:
Do we want `NaN` values to coerce to `0`? (realize that is what was done before, but while were here :-)